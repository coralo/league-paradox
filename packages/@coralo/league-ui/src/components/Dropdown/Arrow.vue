<script>
export default {
  name: 'lu-arrow',
  render (h) {
    return h('svg', {
      attrs: {
        xmlns: 'http://www.w3.org/2000/svg',
        x: '0px',
        y: '0px',
        viewBox: '0 0 100 100',
        enableBackground: 'new 0 0 100 100'
      }
    }, [
      h('polygon', {
        attrs: {
          points: '23.5,30.1 29.8,36.4 50,16.3 70.2,36.4 76.5,30.1 50,3.6'
        }
      }),
      h('polygon', {
        attrs: {
          points: '76.5,69.7 70.2,63.4 49.6,83.9 29.8,64.1 23.5,70.5 49.6,96.6'
        }
      })
    ])
  }
}
</script>
