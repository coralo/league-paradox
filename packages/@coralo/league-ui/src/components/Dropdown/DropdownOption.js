export default {
  name: 'lu-dropdown-option',
  functional: true,
  props: {
    selected: {
      type: Boolean,
      default: false
    },
    focused: {
      type: Boolean,
      default: false
    }
  },
  render (
    h,
    {
      props: { selected, focused },
      data: { staticClass },
      children
    }
  ) {
    return h(
      'div',
      {
        class: staticClass,
        attrs: {
          role: 'option',
          'aria-selected': selected
        }
      },
      children
    )
  }
}
