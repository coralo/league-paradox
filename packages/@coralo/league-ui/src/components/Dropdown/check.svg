<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 19.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 140 110" enable-background="new 0 0 140 110" xml:space="preserve">
<line id="XMLID_2_" fill="#F0E6D2" stroke="#F0E6D2" stroke-width="10" stroke-miterlimit="10" x1="9.6" y1="60" x2="50.4" y2="100"/>
<line id="XMLID_3_" fill="none" stroke="#F0E6D2" stroke-width="10" stroke-miterlimit="10" x1="47.4" y1="96" x2="129" y2="14.4"/>
</svg>
