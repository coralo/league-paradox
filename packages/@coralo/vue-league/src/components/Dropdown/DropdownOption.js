export default {
  name: 'lu-dropdown-option',
  functional: true,
  props: {
    selected: {
      type: Boolean,
      default: false
    },
    focused: {
      type: Boolean,
      default: false
    }
  },
  render (
    h,
    {
      props: { selected, focused },
      data: { class: dynamicClass },
      listeners: { click },
      children
    }
  ) {
    return h(
      'div',
      {
        class: dynamicClass,
        attrs: {
          role: 'option',
          'aria-selected': selected
        },
        on: {
          click
        }
      },
      children
    )
  }
}
