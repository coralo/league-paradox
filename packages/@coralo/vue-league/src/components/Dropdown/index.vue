<template>
  <div
    ref="root"
    :tabindex="disabled ? -1 : tabIndex"
    :class="[
      'dropdown',
      {
        disabled,

      }
    ]"
    :aria-disabled="disabled"
    :aria-expanded="isOpen"
    role="combobox"
    v-on="eventHandlers"
  >
    <div
      :class="[
        'control',
        {
          active: isOpen
        }
      ]"
    >
      {{ `${~selected ? options[selected] : placeholder}` }}
      <arrow class="arrow" />
    </div>
    <div
      v-if="!disabled"
      ref="menu"
      :class="[
        'options',
        {
          hidden: !isOpen
        }
      ]"
      :aria-hidden="!isOpen"
      role="listbox"
    >
      <dropdown-option
        v-for="(option, id) in options"
        :class="[
          'option',
          {
            selected: selected === id
          }
        ]"
        :key="id"
        @click="$emit('select', id)"
      >{{ option }}</dropdown-option>
    </div>
  </div>
</template>

<script>
import Arrow from './Arrow'
import DropdownOption from './DropdownOption'

export default {
  name: 'lu-dropdown',
  components: {
    Arrow,
    DropdownOption
  },
  model: {
    prop: 'selected',
    event: 'select'
  },
  props: {
    options: {
      type: Array,
      default: () => {
        return []
      }
    },
    placeholder: {
      type: String,
      default: 'Select ...'
    },
    tabIndex: {
      type: Number,
      default: 0
    },
    disabled: {
      type: Boolean,
      default: false
    },
    expanded: {
      type: Boolean,
      default: false
    },
    selected: {
      type: Number,
      default: -1
    }
  },
  data () {
    return {
      open: false
    }
  },
  computed: {
    isOpen () {
      return this.disabled ? false : (!!this.open || !!this.expanded)
    },
    eventHandlers () {
      return {
        click: () => this.handleToggle(!this.open)
      }
    }
  },
  methods: {
    handleToggle (toggled) {
      let nextOpen = toggled
      if (nextOpen !== this.open) {
        this.open = nextOpen
      }

      this.$emit('toggle')
    },
    optionClick () {
      console.log('abc')
    }
  }
}
</script>

<style lang="stylus" scoped>
@import '../../assets/styl/variables.styl'

.dropdown
  display block
  position relative
  cursor pointer
  width 100%
  color $lcu-text-dark
  letter-spacing 0.025rem
  outline none

  // line-height: 1;
  &, & div
    box-sizing border-box
    margin 0

  .control
    padding 10px 14px
    background-color rgba(30, 35, 40, 0.5)
    border 1px solid transparent
    border-image linear-gradient(to top, #695625 0%, #A9852D 23%, #B88D35 93%, #C8AA6E 100%) 1
    overflow hidden
    text-overflow ellipsis
    white-space nowrap

    &.active
      background rgba(30, 35, 40, 0.5)
      color #463714
      border 1px solid #463714

    .arrow
      fill #C8AA6E
      position absolute
      top 0
      bottom 0
      right 9px
      margin auto
      cursor pointer
      height 1em

  &:focus .control, .control:hover, .control:focus
    background linear-gradient(to top, rgba(88, 83, 66, 0.5), rgba(30, 35, 40, 0.5))
    border 1px solid transparent
    border-image linear-gradient(to top, #c89b3c, #f0e6d2) 1

  .options
    position absolute
    top 100%
    width 100%
    max-height 250px
    overflow-y auto
    overflow-x hidden
    background #010A13
    border 1px solid transparent
    border-image linear-gradient(to top, #695625, #463714) 1
    z-index 10

    &.hidden
      max-height 0px
      border 0
      overflow hidden

      & > .option
        display none

    .option
      position relative
      color rgb(80.4%, 74.5%, 56.9%)
      cursor pointer
      overflow-x hidden
      padding 10px
      border-top 1px solid rgb(12.2%, 12.9%, 13.7%)
      text-overflow ellipsis
      white-space nowrap

      &.focused, &:focus
        color rgb(94.1%, 90.2%, 82.4%)
        background-color rgb(11.8%, 13.7%, 15.7%)

      &:active
        color rgb(27.5%, 21.6%, 7.8%)
        background-color rgba(30, 35, 40, 0.5)

      &.selected
        padding-right 1.25em
        padding-right calc(1.25em + 10px)

      &.selected::after
        content ''
        position absolute
        right 10px
        top 0
        bottom 0
        background url('./check.svg') center no-repeat
        width 1.25em
        margin auto

  &.disabled
    pointer-events none
    cursor default
    color lcu-text-disabled

    .control
      border-image none
      border-color lcu-text-disabled
      background-color lcu-bg-disabled

      .arrow
        fill lcu-text-disabled
</style>
