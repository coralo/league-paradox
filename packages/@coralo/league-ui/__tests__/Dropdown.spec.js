import { mount, shallowMount } from '@vue/test-utils'
import Arrow from '../src/components/Dropdown/Arrow'
import Dropdown from '../src/components/Dropdown'

describe('Arrow', () => {
  it('render correctly', () => {
    const wrapper = mount(Arrow)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('Dropdown', () => {
  it('render correctly', () => {
    const wrapper = shallowMount(Dropdown)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
