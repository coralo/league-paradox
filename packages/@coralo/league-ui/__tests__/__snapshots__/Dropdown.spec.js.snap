// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow render correctly 1`] = `
<svg xmlns="http://www.w3.org/2000/svg" x="0px" y="0px" viewBox="0 0 100 100" enableBackground="new 0 0 100 100">
  <polygon points="23.5,30.1 29.8,36.4 50,16.3 70.2,36.4 76.5,30.1 50,3.6"></polygon>
  <polygon points="76.5,69.7 70.2,63.4 49.6,83.9 29.8,64.1 23.5,70.5 49.6,96.6"></polygon>
</svg>
`;

exports[`Dropdown render correctly 1`] = `
<div class="dropdown">
  <div class="control">
    <arrow-stub class="arrow"></arrow-stub>
  </div>
</div>
`;
